### ISSUE TITLE
Exception on Initialization - Error creating bean with name 'enableRedisKeyspaceNotificationsInitializer'

### ISSUE THREAD

**Issue by ghsbhatia** [spring-projects/spring-session#653]
spring-session-1.0.2.RELEASE

Steps to Reproduce:
1. Include the following dependency
   
   ```
   <dependency>
       <groupId>org.springframework.session</groupId>
       <artifactId>spring-session</artifactId>
   </dependency>
   ```
2. Start the application when Redis server is not running

Exception:

Error creating bean with name 'enableRedisKeyspaceNotificationsInitializer' defined in class path resource[org/springframework/session/data/redis/config/annotation/web/http/RedisHttpSessionConfiguration.class]: Invocation of init method failed; nested exception is org.springframework.data.redis.RedisConnectionFailureException: Cannot get Jedis connection; nested exception is redis.clients.jedis.exceptions.JedisConnectionException: Could not get a resource from the pool

Root Cause:

A connection is requested from RedisConnectionFactory. If Redis is not running, exception is thrown in the follwing code in RedisHttpSessionConfiguration.java  

```
static class EnableRedisKeyspaceNotificationsInitializer implements InitializingBean {
    private final RedisConnectionFactory connectionFactory;

    private ConfigureRedisAction configure;

    EnableRedisKeyspaceNotificationsInitializer(RedisConnectionFactory connectionFactory, ConfigureRedisAction configure) {
        this.connectionFactory = connectionFactory;
        this.configure = configure;
    }

    public void afterPropertiesSet() throws Exception {
        RedisConnection connection = connectionFactory.getConnection();
        configure.configure(connection);
    }
}
```

Enahancement:
1. Configure ConfigureRedisAction instance only when RedisConnection is available.
2. Change signature of ConfigureRedisAction.configure to accept RedisConnectionFactory rather than RedisConnection. That way, when ConfigureRedisAction.NO_OP is autowired, no attempt is made to create a connection that would be unutilized.

**Comment by DaqingFeng** [comment]
Please check your application.properties config, The  host or password can not allow letter space

**Comment by rwinch** [comment]
I think this makes sense. If anyone is interested in submitting a PR, please mention on here to avoid duplicating efforts.

**Comment by vpavic** [comment]
This was resolved in #657.
